src/lib/useRoller.ts

```diff
@@ -161,69 +155,76 @@ export default function useRoller() {
         new Set<number>()
       );
 
-      console.log(pendingTxs, pendingSpawns);
-
       planets = planets.filter((point: number) => !pendingSpawns.has(point));
 
-      const starInfo = nonces[_point];
-      console.log(starInfo);
+      const starInfo = await api.getPoint(_point);
       const tickets: {
         ticket: string;
         planet: number;
         owner: string;
       }[] = [];
       const requests: Promise<string>[] = [];
 
-      const operator =
-        getSpawnNonce(starInfo, _wallet.address) ||
-        getOwnerNonce(starInfo, _wallet.address);
+      const proxy = isOwnerProxy(starInfo.ownership!, _wallet.address)
+        ? 'own'
+        : isSpawnProxy(starInfo.ownership!, _wallet.address)
+        ? 'spawn'
+        : undefined;
```

Comment:
Looks like this logic is repeated in multiple places. That is probably fine for now, but if we anticipate using it in more places, we should consider extracting it to its own helper.

Reply:
Makes sense, I can make wrappers around these calls